test/types/client_bulk_write.test-d.ts

```diff
@@ -0,0 +1,258 @@
+    document: { name: 'bob', published: 2 }
+  },
+  {
+    namespace: 'db.authors',
+    name: 'replaceOne',
+    filter: { name: 'bob' },
+    replacement: { name: 'ann', published: 2 }
+  },
+  {
+    namespace: 'db.authors',
+    name: 'updateOne',
+    filter: { name: 'bob', published: 2 },
+    update: {}
+  },
+  {
+    namespace: 'db.authors',
+    name: 'updateMany',
+    filter: { name: 'bob', published: 2 },
+    update: {}
+  },
+  { namespace: 'db.authors', name: 'deleteOne', filter: {} },
+  { namespace: 'db.authors', name: 'deleteMany', filter: {} }
+]);
+
+// No schemas - incorrect use - random namespaces, no type checking
+client.bulkWrite([
+  {
+    namespace: 'db.whatever',
+    name: 'insertOne',
+    document: { name: 'bob', randomKey: 2 }
+  },
+  {
+    namespace: 'db.change',
+    name: 'replaceOne',
+    filter: { name: 'bob' },
+    replacement: { name: 2, published: 2 }
+  },
+  {
+    namespace: 'db.it',
+    name: 'updateOne',
+    filter: { name: 'bob', published: new Date() },
+    update: {}
+  },
+  {
+    namespace: 'db.up',
+    name: 'updateMany',
+    filter: { name: 'bob', published: 2 },
+    update: {}
+  },
+  { namespace: 'db.random', name: 'deleteOne', filter: {} },
+  { namespace: 'db.namespace', name: 'deleteMany', filter: {} }
+]);
+
+// Operation names are still type checked when there is no schema
+expectError<ClientBulkWriteModel>({
+  namespace: 'db.author',
+  name: 'insertLots', // Not an operation we support
+  document: { name: 'bob', published: 2 }
+});
+
+type MongoDBSchemas = {
+  'db.books': Book;
+  'db.authors': Author;
+  'db.stores': Store;
+};
+
+expectError<ClientBulkWriteModel<MongoDBSchemas>>({
+  namespace: 'db.author', // Unknown namespace! a typo!
+  name: 'insertOne',
+  document: { name: 'bob', published: 2 }
+});
+
+expectError<ClientBulkWriteModel<MongoDBSchemas>>({
+  namespace: 'db.authors',
+  name: 'insertOne',
+  document: { name: 'bob', published: '' } // Incorrect type for known field
+});
+
+expectError<ClientBulkWriteModel<MongoDBSchemas>>({
+  namespace: 'db.authors',
+  name: 'insertOne',
+  document: { name: 'bob', publish: 2 } // unknown field! typo!
+});
+
+// Defined schemas - all correct use
+client.bulkWrite<MongoDBSchemas>([
+  {
+    namespace: 'db.authors',
+    name: 'insertOne',
+    document: { name: 'bob', published: 2 }
+  },
+  {
+    namespace: 'db.authors',
+    name: 'replaceOne',
+    filter: { name: 'bob' },
+    replacement: { name: 'ann', published: 2 }
+  },
+  {
+    namespace: 'db.authors',
+    name: 'updateOne',
+    filter: { name: 'bob', published: 2 },
+    update: {}
+  },
+  {
+    namespace: 'db.authors',
+    name: 'updateMany',
+    filter: { name: 'bob', published: 2 },
+    update: {}
+  },
+  { namespace: 'db.authors', name: 'deleteOne', filter: {} },
+  { namespace: 'db.authors', name: 'deleteMany', filter: {} }
+]);
+
+// Defined schemas - incorrect use
+expectError(
+  client.bulkWrite<MongoDBSchemas>([
```

Comment:
I don't think this works - won't this fail if only one of the incorrect usages is causing an error?  I think each "incorrect" case needs a separate expectError

Reply:
It works but yea good catch we wouldn't catch the individual parts of this failing